src/aws-cpp-sdk-core/source/auth/GeneralHTTPCredentialsProvider.cpp

```diff
@@ -121,7 +121,11 @@ bool GeneralHTTPCredentialsProvider::ShouldCreateGeneralHTTPProvider(const Aws::
                 shouldAllow = !addresses.empty();
                 hostResolved = true;
               }
-              hostResolverCV.notify_one();
+              else
+              {
+                std::unique_lock<std::mutex> lock(hostResolverMutex);
+                hostResolverCV.notify_one();
```

Comment:
wait
CV should be still notified in success case
otherwise there might be a rare race condition issue

when on the main thread,
```
if (!hostResolved) {
```
is not resolved and goes into if branch
but then before hitting the the line, it get resolved
```
hostResolverCV.wait_for(lock, std::chrono::milliseconds(1000));
```
resulting in an unnecessary 1s wait (which is here just to avoid infinite wait in case of any other issues to be honest).

Reply:
Ok, added it.

Reply:
perfect, thank you, will merge soon!